sqlglot/dialects/bigquery.py

```diff
@@ -91,3 +91,12 @@ class Generator(Generator):
             exp.DataType.Type.BOOLEAN: "BOOL",
             exp.DataType.Type.TEXT: "STRING",
         }
+
+        def union_sql(self, expression):
```

Comment:
I wonder if we should be overriding `set_operation` instead of `union_sql`. I don't know bigquery that well, but from the link you shared, looks like we might want the same for EXCEPT and INTERSECT.

Also, if we override `set_operation` there will be less duplication of Generator code.

Reply:
Ah, EXCEPT and INTERSECT already add the DISTINCT token. Hm.

I'll let @tobymao comment on this one. I wonder if we want to add some other method/attribute on the base Generator so we don't have to override the entire union_sql method.

Reply:
Thanks for taking a look so quickly!

It looks like BigQuery doesn't support EXCEPT or INTERSECT without DISTINCT, so maybe we should throw an error if you try to render one that doesn't have `args['distinct'] == True`?

Reply:
How about three new methods, `intersect_op`, `except_op`, `union_op`.

The defaults would look like:

```python
def union_op(self, expression):
    return f"UNION{' DISTINCT' if expression.args.get('distinct') else ' ALL'}"
```

Just a smaller surface area to override.

> so maybe we should throw an error 

Check out `Generator.unsupported`